pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1660,3 +1660,34 @@ func MutateShootETCDEncryptionKeyRotation(shoot *gardencorev1beta1.Shoot, f func
 
 	f(shoot.Status.Credentials.Rotation.ETCDEncryptionKey)
 }
+
+// IsValidShootOperation checks whether the provided operation annotation value is valid. It returns two bools: The
+// first value specifies whether it's valid, and the second bool value specifies whether the gardener-apiserver removes
+// the annotation before the Shoot is persisted to etcd.
+func IsValidShootOperation(operation string, lastOperation *gardencorev1beta1.LastOperation) (isValid bool, removedBeforePersisted bool) {
+	if lastOperation != nil && lastOperation.State == gardencorev1beta1.LastOperationStateFailed {
+		if operation == v1beta1constants.ShootOperationRetry {
+			return true, true
+		}
+	} else {
+		switch operation {
+		case v1beta1constants.GardenerOperationReconcile:
+			return true, true
+
+			// TODO(rfranzke): Handle new operation annotations introduced by
+			// - https://github.com/gardener/gardener/pull/6021
```

Comment:
```suggestion
```

#6021 is merged, we can add to the list the operation annotations related to the etcd encryption key.